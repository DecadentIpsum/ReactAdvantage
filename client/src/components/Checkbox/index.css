.checkbox-wrapper {
  display: flex;
  align-items: center;
  cursor: pointer; }

.checkbox-label {
  font-size: 1em;
  color: #B3B3AB;
  cursor: pointer; }

body .ui-chkbox-box.ui-state-active,
body .ui-radiobutton-box.ui-state-active {
  background: #5A88C3;
  border-color: #5A88C3; }
