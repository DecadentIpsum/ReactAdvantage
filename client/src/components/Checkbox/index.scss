@import "global-styles/common.scss";

.checkbox-wrapper {
    display: flex;
    align-items: center;
    cursor: pointer;
}

.checkbox-label {
    font-size: 1em;
    color: $grayShade;
    cursor: pointer;
    margin-left: 10px;
}

body {
    .ui-chkbox-box.ui-state-active,
    .ui-radiobutton-box.ui-state-active {
        background: $primary;
        border-color: $primary;
    }
}
