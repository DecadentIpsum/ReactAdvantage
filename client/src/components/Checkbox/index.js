import React, { Component } from 'react';
import { Checkbox as PrimeCheckbox } from 'primereact/components/checkbox/Checkbox';

import './index.css';

export default class Checkbox extends Component {
    renderLabel() {
        const { label, inputId } = this.props;

        return (
            <label htmlFor={inputId} className="checkbox-label">
                {label}
            </label>
        );
    }

    render() {
        const { label, ...restProps } = this.props;

        return (
            <div className="checkbox-wrapper">
                <PrimeCheckbox {...restProps} />
                {label && this.renderLabel()}
            </div>
        );
    }
}
