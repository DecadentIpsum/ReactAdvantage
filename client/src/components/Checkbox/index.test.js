import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import Checkbox from './index';

test('checkbox without label renders correctly', () => {
    const component = shallow(<Checkbox />);

    expect(component).toMatchSnapshot();
});

test('checkbox with label renders correctly', () => {
    const component = shallow(
        <Checkbox
            label="test"
            id="test"
        />
    );

    expect(component).toMatchSnapshot();
});
