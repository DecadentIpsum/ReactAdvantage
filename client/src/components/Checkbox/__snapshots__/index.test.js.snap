// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox with label renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox
    id="test"
    label="test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Checkbox
          checked={false}
          className={null}
          disabled={false}
          id="test"
          inputId={null}
          name={null}
          onChange={null}
          onContextMenu={null}
          onMouseDown={null}
          readOnly={false}
          style={null}
          value={null}
/>,
        <label
          className="checkbox-label"
          htmlFor={undefined}
>
          test
</label>,
      ],
      "className": "checkbox-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "checked": false,
          "className": null,
          "disabled": false,
          "id": "test",
          "inputId": null,
          "name": null,
          "onChange": null,
          "onContextMenu": null,
          "onMouseDown": null,
          "readOnly": false,
          "style": null,
          "value": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": "checkbox-label",
          "htmlFor": undefined,
        },
        "ref": null,
        "rendered": "test",
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Checkbox
            checked={false}
            className={null}
            disabled={false}
            id="test"
            inputId={null}
            name={null}
            onChange={null}
            onContextMenu={null}
            onMouseDown={null}
            readOnly={false}
            style={null}
            value={null}
/>,
          <label
            className="checkbox-label"
            htmlFor={undefined}
>
            test
</label>,
        ],
        "className": "checkbox-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": false,
            "className": null,
            "disabled": false,
            "id": "test",
            "inputId": null,
            "name": null,
            "onChange": null,
            "onContextMenu": null,
            "onMouseDown": null,
            "readOnly": false,
            "style": null,
            "value": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "className": "checkbox-label",
            "htmlFor": undefined,
          },
          "ref": null,
          "rendered": "test",
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`checkbox without label renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Checkbox
          checked={false}
          className={null}
          disabled={false}
          id={null}
          inputId={null}
          name={null}
          onChange={null}
          onContextMenu={null}
          onMouseDown={null}
          readOnly={false}
          style={null}
          value={null}
/>,
        undefined,
      ],
      "className": "checkbox-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "checked": false,
          "className": null,
          "disabled": false,
          "id": null,
          "inputId": null,
          "name": null,
          "onChange": null,
          "onContextMenu": null,
          "onMouseDown": null,
          "readOnly": false,
          "style": null,
          "value": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Checkbox
            checked={false}
            className={null}
            disabled={false}
            id={null}
            inputId={null}
            name={null}
            onChange={null}
            onContextMenu={null}
            onMouseDown={null}
            readOnly={false}
            style={null}
            value={null}
/>,
          undefined,
        ],
        "className": "checkbox-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": false,
            "className": null,
            "disabled": false,
            "id": null,
            "inputId": null,
            "name": null,
            "onChange": null,
            "onContextMenu": null,
            "onMouseDown": null,
            "readOnly": false,
            "style": null,
            "value": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
